gomint-server/src/main/java/io/gomint/server/world/WorldAdapter.java

```diff
@@ -149,8 +173,12 @@ protected WorldAdapter(GoMintServer server, File worldDir, String name) {
         this.asyncChunkTasks = new LinkedBlockingQueue<>();
         this.chunkPackageTasks = new ConcurrentLinkedQueue<>();
         this.entitySpawner = new EntitySpawner(this);
-        this.startAsyncWorker(server.scheduler());
+        this.syncTaskManager = new SyncTaskManager();
+        this.syncSchedulerAdapter = new SyncSchedulerAdapter(this.syncTaskManager);
+        this.startAsyncWorker(server.asyncScheduler());
         this.initGamerules();
+        this.thread = new Thread(this::tickLoop, "GoMint World Thread " + worldDir.getName());
```

Comment:
This thread should come out of a threadpool so when closing the world this thread can be assigned to another world

Reply:
I thought thread pools are only relevant for short-living threads.

Reply:
Thread pools are relevant for all sorts of things, mostly to ensure that we can keep proper track of it on shutdown. There is a thread reaper system which seperates the display of system relevant thread by its thread group: https://github.com/gomint/gomint/blob/8523fd0a0be896eb8fbb4a802b3059ddb5e7498d/gomint-server/src/main/java/io/gomint/server/GoMintServer.java#L626 there should be a group named "gomint-worlds" from which all threads come so we can warn properly

Reply:
Seems to me like the only place thread groups are used in whole gomint is at that point. Seems to me like thread grouping (not pooling) is not a thing yet in gomint.

Reply:
it will be in the future (to ensure that async tasks attached to a plugin get terminated when the plugin unloads)

Reply:
resolved?